fastlane/Fastfile

```diff
@@ -1336,13 +1303,16 @@ end
 # Kicks off a Buildkite build
 # -----------------------------------------------------------------------------------
 def trigger_buildkite_release_build(branch:, beta:)
-  buildkite_trigger_build(
+  build_url = buildkite_trigger_build(
     buildkite_organization: 'automattic',
     buildkite_pipeline: 'woocommerce-ios',
     branch: branch,
     environment: { BUILDKITE_BETA_RELEASE: beta },
     pipeline_file: 'release-builds.yml'
   )
+
+  message = "This build triggered a build on <code>#{branch}</code>:<br>- #{build_url}"
+  buildkite_annotate(style: 'info', context: 'trigger-release-build', message: message) if is_ci
```

Comment:
Side note and reminder to self that we really need to try the idea of using `buildkite-agent pipeline upload release-builds.yml` to append new steps to the existing build as opposed to triggering a trampoline build. iinm only change required for this to work would be to ensure the steps in that `release-builds.yml` pipeline check out the `branch` branch before they run, since the branch on which the current CI build is running might be a different branch.

---

Also note to self that we should get rid of that ugly workaround of using `BUILDKITE_` prefixed environment variable here, not only because that is not necessary and can be avoided, but also because `BUILDKITE_`-prefixed env vars are reserved for Buildkite internal use and polluting that namespace with our own env vars might break Buildkite's behavior unexpectedly at some point.

See https://github.com/bloom/DayOne-Apple/pull/24228#discussion_r1719712518 for more explanations and the solution to avoid it

Reply:
> Side note and reminder to self that we really need to try the idea of using buildkite-agent pipeline upload release-builds.yml

Good one 👍  perhaps once I'm done with this initial set of ReleasesV2 improvements I can try that out.

> 🎗️ Also note to self that we should get rid of that ugly workaround of using BUILDKITE_ prefixed environment variable

👍